ldk/goja/src/whisper/index.ts

```diff
@@ -0,0 +1,255 @@
+export enum WhisperComponentType {
+  BOX = 'box',
+  BUTTON = 'button',
+  CHECKBOX = 'checkbox',
+  COLLAPSE_BOX = 'collapseBox',
+  DIVIDER = 'divider',
+  LINK = 'link',
+  LIST_PAIR = 'listPair',
+  MARKDOWN = 'markdown',
+  MESSAGE = 'message',
+  NUMBER = 'number',
+  PASSWORD = 'password',
+  RADIO_GROUP = 'radioGroup',
+  SELECT = 'select',
+  TELEPHONE = 'telephone',
+  TEXT_INPUT = 'textInput',
+}
+
+export enum Alignment {
+  CENTER = 'center',
+  LEFT = 'left',
+  RIGHT = 'right',
+  SPACE_AROUND = 'space_around',
+  SPACE_EVENLY = 'space_evenly',
+}
+
+export enum Direction {
+  HORIZONTAL = 'horizontal',
+  VERTICAL = 'vertical',
+}
+
+export enum TextAlign {
+  CENTER = 'center',
+  LEFT = 'left',
+  RIGHT = 'right',
+}
+
+export enum Urgency {
+  ERROR = 'error',
+  NONE = 'none',
+  SUCCESS = 'success',
+  WARNING = 'warning',
+}
+
+export interface WhisperComponent {
+  id?: string;
+  type: string;
+}
+
+export interface Button extends WhisperComponent {
```

Comment:
Right now this type declaration would support an object like:
```
{
  id: 'button',
  type: 'text',
  label: 'This should be a button but it isn't',
}
```

That'd probably not be great so instead I'd make the `WhisperComponent` a generic type, so it'd look like this:
```
export interface WhisperComponent<T extends WhisperComponentType> {
  id?: string;
  type: T;
}

export interface Button extends WhisperComponent<WhisperComponentType.Button> {
  ...
}
```